dolphin/drivers/api.py

```diff
@@ -31,12 +31,18 @@ def get_storage_registry():
         """Show register parameters which the driver needs."""
         pass
 
-    def register_storage(self, context, access_info):
-        """Discovery a storage system with access information."""
-        access_info['storage_id'] = six.text_type(uuidutils.generate_uuid())
-        driver = self.driver_manager.create_driver(context, **access_info)
+    def discover_storage(self, context, access_info):
+        """Discover a storage system with access information and create driver instance ."""
+
+        if 'storage_id' not in access_info:
+            access_info['storage_id'] = six.text_type(uuidutils.generate_uuid())
+        else:
+            # Already registered storage , remove driver and create driver instance \
+            # again with new access_info
+            self.driver_manager.remove_driver(context,access_info['storage_id'])
```

Comment:
Consider multiple nodes for distribution system, how to remove drivers from other nodes? If not removed, they may cause failure in the future on other nodes.

Reply:
Yes, we need to analyse impact of  all multi node usecases for all driver manager interfaces.
Created an Issue for this https://github.com/katurii/dolphin/issues/50